### PR TITLE
Add PartialGroupDMChannel class

### LINKED ISSUE #3345: Using `Client.fetchInvite` to fetch Group DM Invite metadata results in a missing (null) channel property on the returned Invite object
**Please describe the problem you are having in as much detail as possible:**

When using `Client.fetchInvite` to fetch metadata about a Group DM Invite code, the returned Invite object has the `channel` property set to `null`, even though the API returns a partial DM Group Channel object.
Additionally, DJS will emit the debug event `Failed to find guild, or unknown type for channel <channel ID> 3`.
This is probably because DJS doesn't support Group DMs anymore in v12 and as such it doesn't know how to resolve the channel type 3.

Currently the documentation for the `Invite` object states that `channel` is a non-nullable property, which appears to be wrong in the case of Group Invites. If this is intended behavior the docuentation and typings should be adjusted to reflect that `channel` might be null.
However I've also asked on the Support Server and we came to the conclusion that it would be nice to have a partial channel object for Group Invites; after all the channel data is included in the API response ([relevant message link](https://discordapp.com/channels/222078108977594368/222078374472843266/589428838916816897)).

**Reproduction steps:**

- Generate an invite link to a Group DM
- Use `fetchInvite` to... fetch it
- See that the returned Invite object has `channel: null`

**Further details:**
- discord.js version: v12 on master as of 6cd4c27faed8b3f1b9da41a87b7922cd286313b9
- Node.js version: v10.12.0
- Priority this issue should have – please be realistic and elaborate if possible: Low

- [x] I have also tested the issue on latest master, commit hash: 6cd4c27faed8b3f1b9da41a87b7922cd286313b9

### PR DESCRIPTION
**Pull request by NotSugden**
**Please describe the changes this PR makes and why it should be merged:**
This PR fixes #3345 by adding a Partial Group DM Channel class for the partial data that is received when fetching an invite to a Group DM

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/Channel.js [hunk 1]
```diff
@@ -82,6 +83,7 @@ class Channel extends Base {
    *   .catch(console.error);
    */
   delete() {
+    if (this.type === 'group') return Promise.reject(new Error('DELETE_GROUP_DM_CHANNEL'));
```
Comment (SpaceEEC) [inline src/structures/Channel.js]: Override the method in the deriving class (`PartialGroupDMChannel`) and throw the error there.

File: src/structures/Channel.js [hunk 2]
```diff
@@ -90,15 +92,26 @@ class Channel extends Base {
    * @returns {Promise<Channel>}
    */
   fetch() {
+    if (this.type === 'group') return Promise.reject(new Error('FETCH_GROUP_DM_CHANNEL'));
```
Comment (SpaceEEC) [inline src/structures/Channel.js]: Override the method in the deriving class (`PartialGroupDMChannel`) and throw the error there.

File: src/structures/PartialGroupDMChannel.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+const Channel = require('./Channel');
+
+/**
+ * Represents a Partial Group DM Channel on Discord.
+ * @extends {Channel}
+ */
+class PartialGroupDMChannel extends Channel {
+  constructor(client, data) {
+    super(client, data);
+
```
Comment (SpaceEEC) [inline src/structures/PartialGroupDMChannel.js]: Group dm channels have names. 👀

Comment (NotSugden) [reply src/structures/PartialGroupDMChannel.js]: oops

**Pull request merged**